runway/blueprints/staticsite/auth_at_edge.py

```diff
@@ -0,0 +1,434 @@
+                   ):
+        # type: (...) -> Dict[str, Any]
+        """Creates a lambda function and it's version.  # noqa
+
+        Args:
+            title (str): The name of the function in PascalCase
+            description (str): Description to be displayed in the
+                lambda panel
+            handle (str): The underscore separated representation
+                of the name of the lambda. This handle is used to
+                determine the handler for the lambda as well as
+                identify the correct Code hook_data information.
+            role (IAM.Role): The Lambda Execution Role
+        """
+        function = self.get_auth_at_edge_lambda(
+            title,
+            description,
+            handle,
+            role
+        )
+
+        return {
+            'function': function,
+            'version': self.add_version(title, function)
+        }
+
+    def get_auth_at_edge_lambda(self,
+                                title,   # type: str
+                                description,   # type: str
+                                handler,  # type: str
+                                role  # type: iam.Role
+                                ):
+        # type: (...) -> awslambda.Function
+        """Create an Auth@Edge lambda resource.
+
+        Args:
+            name (str): The name of the function in PascalCase
+            description (str): Description to be displayed in the
+                lambda panel
+            handle (str): The underscore separated representation
+                of the name of the lambda. This handle is used to
+                determine the handler for the lambda as well as
+                identify the correct Code hook_data information.
+            role (IAM.Role): The Lambda Execution Role
+        """
+        return self.template.add_resource(
+            awslambda.Function(
+                title,
+                Code=self.context.hook_data['aae_lambda_config'].get(handler),
+                Description=description,
+                Handler='__init__.handler',
+                Role=role.get_att("Arn"),
+                Runtime='python3.7'
+            )
+        )
+
+    def add_version(self,
+                    title,  # type: str
+                    lambda_function  # type: awslambda.Function
+                    ):
+        # type: (...) -> awslambda.Version
+        """Create a version association with a Lambda@EDGE function.
+
+        In order to ensure different versions of the function
+        are appropriately uploaded a hash based on the code of the
+        lambda is appended to the name. As the code changes so
+        will this hash value.
+
+        Args:
+            title (str): The name of the function in PascalCase
+            role (awslambda.Function): The Lambda function
+        """
+        code_hash = hashlib.md5(
+            str(lambda_function.properties['Code'].to_dict()).encode()
+        ).hexdigest()
```

Comment:
The `Code.S3Key` already contains a hash based on file contents that we can grab instead of calculating a different hash. may be more cohesive.

Reply:
Gonna pull in @troyready on this one. I used his code originally from the static site. Totally fine changing this, but I'll have to change both spots. That going to work ok @troyready?

Reply:
May be worth a `# TODO` then and we can do it in `2.0`? it's not _breaking_ but will cause an unnecessary update to the lambda function so im torn.

Reply:
Added the @TODO